### PR TITLE
add neural sequence tagging model for QA

### PR DESCRIPTION
**Pull request by pengli09**
Implement model proposed in _Peng Li, Wei Li, Zhengyan He, Xuguang Wang, Ying Cao, Jie Zhou, and Wei Xu. Dataset and Neural Recurrent Sequence Labeling Model for Open-Domain Factoid Question Answering. [arXiv:1607.06275](https://arxiv.org/abs/1607.06275)_.

### CODE REVIEW

File: neural_seq_qa/config.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+class CommonConfig(object):
+    def __init__(self):
+        # network size:
+        # dimension of the question LSTM
+        self.q_lstm_dim = 64
+        # dimension of the attention layer
+        self.latent_chain_dim=64
+        # dimension of the evidence LSTMs
+        self.e_lstm_dim=64
```
Comment (lcy-seso) [inline neural_seq_qa/config.py]: This file is not formatted.

File: neural_seq_qa/infer.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+import os
+import sys
+import argparse
+
+import paddle.v2 as pd
```
Comment (lcy-seso) [inline neural_seq_qa/infer.py]: I prefer to use `import paddle.v2 as paddle` to keep consistent with all the other PaddlePaddle demos in models and book.

File: neural_seq_qa/network.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+    :type input: LayerOutput
+    :param word_vec_dim: dimension of the word embeddings
+    :type word_vec_dim: int
+    :param wordvecs: word embedding matrix
+    :type wordvecs: numpy array
+    :return: embedding
+    :rtype: LayerOutput
+    """
+    return pd.layer.embedding(
+                name=name,
```
Comment (lcy-seso) [inline neural_seq_qa/network.py]: To follow current PaddlePaddle's style, if the name is not explicitly used, it is set automatically by PaddlePaddle.

Currently, in all the PaddlePaddle's released demos, except the data layer, all the other layers do not need the name.

**Review by lcy-seso — COMMENTED** [review]
some simple comments first.

File: neural_seq_qa/config.py (hunk 1 above)
Comment (pengli09) [reply neural_seq_qa/config.py]: done

File: neural_seq_qa/infer.py (hunk 2 above)
Comment (pengli09) [reply neural_seq_qa/infer.py]: All `pd` have been changed to `paddle`

File: neural_seq_qa/network.py (hunk 3 above)
Comment (pengli09) [reply neural_seq_qa/network.py]: The names of the layers have been removed.

File: neural_seq_qa/README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+# Training
+
+Training can be launched using the following command:
+
+```bash
+PYTHONPATH=data/evaluation:$PYTHONPATH python train.py 2>&1 | tee train.log
+```
+# Validation and Test
+
+WebQA provoides two versions of validation and test sets.  Automatic valiation and test can be lauched by
```
Comment (ranqiu92) [inline neural_seq_qa/README.md]: provoides -> provides,  valiation ->validation

File: neural_seq_qa/network.py [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+    reader.E_IDS_STR: reader.E_IDS,
+    reader.QE_COMM_STR: reader.QE_COMM,
+    reader.EE_COMM_STR: reader.EE_COMM,
+    reader.LABELS_STR: reader.LABELS
+}
+
+
+def get_embedding(input, word_vec_dim, wordvecs):
+    """
+    Defined word embedding
```
Comment (ranqiu92) [inline neural_seq_qa/network.py]: Defined -> Define

File: neural_seq_qa/network.py [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+    :type q_lstm_dim: int
+    :param latent_chain_dim: dimension of the attention layer
+    :type latent_chain_dim: int
+    :param word_vec_dim: dimension of the word embeddings
+    :type word_vec_dim: int
+    :param drop_rate: dropout rate
+    :type drop_rate: float
+    :param wordvecs: word embedding matrix
+    :type wordvecs: numpy array
+    :param default_init_std: default initial std
```
Comment (ranqiu92) [inline neural_seq_qa/network.py]: Please use the full name of std in all files.

**Review by ranqiu92 — CHANGES_REQUESTED** [review]

File: neural_seq_qa/README.md (hunk 4 above)
Comment (pengli09) [reply neural_seq_qa/README.md]: done

File: neural_seq_qa/network.py (hunk 5 above)
Comment (pengli09) [reply neural_seq_qa/network.py]: done

File: neural_seq_qa/network.py (hunk 6 above)
Comment (pengli09) [reply neural_seq_qa/network.py]: done

**Review by ranqiu92 — APPROVED** [review]

**Review by lcy-seso — CHANGES_REQUESTED** [review]
Sorry, just to cancel the misoperation of clicking approve. Please first ignore this request changes.

**Review by lcy-seso — APPROVED** [review]
LGTM. A question, can we provide a trained model so that other users can directly download it and try it.

**Comment by pengli09** [comment]
@lcy-seso This is a good idea. Shall we provide it in another place? Otherwise, this repo will become very large in the future.

**Comment by lcy-seso** [comment]
Yes~ We do have a place to save the trained model (You can give the model to me, I have the account). We have this example https://github.com/PaddlePaddle/models/blob/develop/image_classification/models/model_download.sh#L29. We can only add the download/evaluation script into the models repo, and update the README.

**Pull request merged**